config/api/models/System.php

```diff
@@ -32,28 +32,23 @@
 		'slugs'        => fn () => Str::$language,
 		'title'        => fn () => $this->site()->title()->value(),
 		'translation' => function () {
-			if ($user = $this->user()) {
-				$translationCode = $user->language();
-			} else {
-				$translationCode = $this->kirby()->panelLanguage();
-			}
+			$code   = $this->user()?->language();
+			$code ??= $this->kirby()->panelLanguage();
```

Comment:
To be honest I think it's a step backwards here. It's much shorter, but if I only looked at the new code, I'd need a lot more time to parse the functionality mentally. Like: "How can `$code` ever be null in the second line? Ah, because of the null-safe operator. But what does this mean for the execution of the second line? Ah, only when `user()` returns `null`." Whereas the old code is really explicit about the two different cases.

Reply:
Very interesting. For me it's literally exact the opposite: if-else always requires so much more brain capacity for me as my mind needs to keep track of two parallel contexts that keep developing and seeing how to merge them together afterwards. Whereas the usage of `??=` turns it into a one-dimensional order: "if it's still null, then fill it with the following". Reducing for my mind a lot of complexity. But I see your argument, would just go the way the majority of the team feels best with.

Reply:
I like the one-liner. If it gets too long, we could also do:

```php
$code = $this->user()?->language() ??
    $this->kirby()->panelLanguage();
```

Reply:
My vote is for one liner too 👍